src/Misc/layoutroot/run.cmd

```diff
@@ -13,21 +13,22 @@ if defined VERBOSE_ARG (
 rem Unblock files in the root of the layout folder. E.g. .cmd files.
 powershell.exe -NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command "$VerbosePreference = %VERBOSE_ARG% ; Get-ChildItem -LiteralPath '%~dp0' | ForEach-Object { Write-Verbose ('Unblock: {0}' -f $_.FullName) ; $_ } | Unblock-File | Out-Null"
 
-if /i "%~1" equ "localRun" (
-    rem ********************************************************************************
-    rem Local run.
-    rem ********************************************************************************
-    "%~dp0bin\Runner.Listener.exe" %*
-) else (
-  rem ********************************************************************************
-  rem Run.
-  rem ********************************************************************************
-  "%~dp0bin\Runner.Listener.exe" run %*
 
-  rem Return code 4 means the run once runner received an update message.
-  rem Sleep 5 seconds to wait for the update process finish and run the runner again.
-  if ERRORLEVEL 4 (
-    timeout /t 5 /nobreak > NUL
-    "%~dp0bin\Runner.Listener.exe" run %*
-  )
+rem ********************************************************************************
+rem Run.
+rem ********************************************************************************
+
+:launch_helper
+call "%~dp0bin\run-helper.cmd" %*
+
+rem using `if %ERRORLEVEL% EQU N` insterad of `if ERRORLEVEL N`
+rem `if ERRORLEVEL N` means: error level is N or MORE
+  
+if %ERRORLEVEL% GEQ 2 (
+  echo "Restart runner after it exited with return code '%ERRORLEVEL%'"
+  goto :launch_helper
```

Comment:
should we pass the last exit code to the helper, so it can determine what to do next. Maybe as an env var (so it doesnt interfere with command line args).

Reply:
discussing offline.... ok maybe if we only restart on exit code 2. The scenarios i'm concerned about is some kind of crash and then we get unexpected exit code.

Reply:
food for thought

Reply:
So, based on a comment `@TingluoHuang` left, I simplified the contract between `run` and `helper`, so the helper only returns 0 for exit and 1 for restart.

I understand how `helper` now goes agains exit code conventions (1 for error), but since the helper is not meant to be run in isolation, I think it's fine. Lmk what you think @ericsciple